runtime/mux.go

```diff
@@ -48,11 +48,15 @@ var encodedPathSplitter = regexp.MustCompile("(/|%2F)")
 // A HandlerFunc handles a specific pair of path pattern and HTTP method.
 type HandlerFunc func(w http.ResponseWriter, r *http.Request, pathParams map[string]string)
```

Comment:
@nikitaksv what is the need of pathParams here, can't we just implement a http.HandlerFunc? because of this i'm not able to use normal http middlewares

Reply:
@johanbrandhorst can you please help me?

Reply:
You can wrap the `runtime.ServeMux` in normal HTTP middleware, see https://grpc-ecosystem.github.io/grpc-gateway/docs/operations/logging/.